lib/features/mailbox_dashboard/domain/usecases/get_unread_spam_emails_interactor.dart

```diff
@@ -0,0 +1,50 @@
+
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/unsigned_int.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox_filter_condition.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/model/unread_spam_emails_request.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/model/unread_spam_emails_response.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/repository/spam_report_repository.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/state/get_unread_spam_emails_state.dart';
+
+class GetUnreadSpamEmailsInteractor {
```

Comment:
Change the name of use-case, UI state, failure

`InvalidSpamReportCondition extend FeatureFailure`